src/main/java/org/jenkinsci/plugins/workflow/actions/TaskInfoAction.java

```diff
@@ -0,0 +1,94 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import hudson.model.Queue;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public abstract class TaskInfoAction extends InvisibleAction implements PersistentAction {
+    private static final long serialVersionUID = 1;
+
+    public enum QueueState {
+        QUEUED,
+        CANCELLED,
+        LAUNCHED,
+        UNKNOWN
+    }
+
+    /**
+     * Used to identify the {@link org.jenkinsci.plugins.workflow.steps.StepContext} in the task, so that
+     * its status can be identified.
+     */
+    protected int taskContextHashcode;
```

Comment:
:bug: Remove this and allow the Impl to define how you're storing this -- because the abstract version isn't using it, and shouldn't be tied to this in case we change it later